pandas/core/indexes/base.py

```diff
@@ -151,6 +151,11 @@ class Index(IndexOpsMixin, StringAccessorMixin, PandasObject):
 
     _engine_type = libindex.ObjectEngine
 
+    _accessors = frozenset(['dt', 'str', 'cat'])
```

Comment:
ah, but they are not yet really added as accessor (so no user visible change), only 'removed' from `__dir__` (but not really removed since they are not there yet), correct? 
In that case not that important.

I suppose we already have tests to check that `'str'` is only present on the correct index types?